test/system/150-login.bats

```diff
@@ -0,0 +1,339 @@
+
+    # Yay! Pull it back
+    run_podman pull --tls-verify=false \
+               --creds ${PODMAN_LOGIN_USER}:${PODMAN_LOGIN_PASS} \
+               localhost:${PODMAN_LOGIN_REGISTRY_PORT}/$destname
+
+    # Compare to original busybox
+    run_podman inspect --format '{{.Id}}' $destname
+    is "$output" "$id_busybox" "Image ID of pulled image == busybox"
+
+    run_podman rmi busybox $destname
+}
+
+# END   primary podman login/push/pull tests
+###############################################################################
+# BEGIN cooperation with skopeo
+
+# Skopeo helper - keep this separate, so we can test with different
+# envariable settings
+function _test_skopeo_credential_sharing() {
+    if ! type -p skopeo; then
+        skip "skopeo not available"
+    fi
+
+    registry=localhost:${PODMAN_LOGIN_REGISTRY_PORT}
+
+    run_podman login "$@" --tls-verify=false \
+               --username ${PODMAN_LOGIN_USER} \
+               --password ${PODMAN_LOGIN_PASS} \
+               $registry
+
+    destname=skopeo-ok-$(random_string 10 | tr A-Z a-z)-ok
+    echo "# skopeo copy ..."
+    run skopeo copy "$@" \
+        --format=v2s2 \
+        --dest-tls-verify=false \
+        containers-storage:$IMAGE \
+        docker://$registry/$destname
+    echo "$output"
+    is "$status" "0" "skopeo copy - exit status"
+    is "$output" ".*Copying blob .*"     "output of skopeo copy"
+    is "$output" ".*Copying config .*"   "output of skopeo copy"
+    is "$output" ".*Writing manifest .*" "output of skopeo copy"
+
+    echo "# skopeo inspect ..."
+    run skopeo inspect "$@" --tls-verify=false docker://$registry/$destname
+    echo "$output"
+    is "$status" "0" "skopeo inspect - exit status"
+
+    got_name=$(jq -r .Name <<<"$output")
+    is "$got_name" "$registry/$dest_name" "skopeo inspect -> Name"
+
+    # Now try without a valid login; it should fail
+    run_podman logout "$@" $registry
+    echo "# skopeo inspect [with no credentials] ..."
+    run skopeo inspect "$@" --tls-verify=false docker://$registry/$destname
+    echo "$output"
+    is "$status" "1" "skopeo inspect - exit status"
+    is "$output" ".*: unauthorized: authentication required" \
+       "auth error on skopeo inspect"
+}
+
+@test "podman login - shares credentials with skopeo - default auth file" {
+    if is_rootless; then
+        if [ -z "${XDG_RUNTIME_DIR}" ]; then
+            skip "skopeo does not match podman when XDG_RUNTIME_DIR unset; #823"
+        fi
+    fi
+    _test_skopeo_credential_sharing
+}
+
+@test "podman login - shares credentials with skopeo - via envariable" {
+    skip "skopeo does not yet support REGISTRY_AUTH_FILE; #822"
+    authfile=${PODMAN_LOGIN_WORKDIR}/auth-$(random_string 10).json
+    rm -f $authfile
+
+    REGISTRY_AUTH_FILE=$authfile _test_skopeo_credential_sharing
+    rm -f $authfile
+}
+
+@test "podman login - shares credentials with skopeo - via --authfile" {
+    # Also test that command-line --authfile overrides envariable
```

Comment:
should you skip this one right off the bat too?

Reply:
Haha! No, that's actually deliberate: since `--authfile=/path` should always override environment, `$REGISTRY_AUTH_FILE` should be ignored by both podman and skopeo - hence it shouldn't matter whether or not skopeo implements that or not. And if/when skopeo _does_ implement it, if for any reason it gets implemented wrong such that environment overrides command-line, this test will catch it.

Does that make sense? I had hoped that the test comment would make it clear without having to go into subtleties, but should I update the comment, and if so, can you suggest an improvement?

Reply:
no worries, just me apparently reviewing too quickly.